packages/@aws-cdk/aws-events-targets/lib/sqs.ts

```diff
@@ -0,0 +1,71 @@
+import events = require('@aws-cdk/aws-events');
+import iam = require('@aws-cdk/aws-iam');
+import sqs = require('@aws-cdk/aws-sqs');
+
+const fifoQualifier = "fifo";
+
+/**
+ * Customize the SQS Queue Event Target
+ */
+export interface SqsQueueProps {
+
+  /**
+   * The messageGroupId for the queue
+   *
+   * @default empty string
+   */
+  readonly messageGroupId?: string;
+
+  /**
+   * The message to send to the queue
+   *
+   * @default the entire CloudWatch event
+   */
+  readonly message?: events.RuleTargetInput;
+
+}
+
+/**
+ * Use an SQS Queue as a target for AWS CloudWatch event rules.
+ *
+ * @example
+ *
+ *    // publish to an SQS queue every time code is committed
+ *    // to a CodeCommit repository
+ *    repository.onCommit(new targets.SqsQueue(queue));
+ *
+ */
+export class SqsQueue implements events.IRuleTarget {
+
+  constructor(public readonly queue: sqs.IQueue, private readonly props: SqsQueueProps = {}) {
+  }
+
+  /**
+   * Returns a RuleTarget that can be used to trigger this SQS queue as a
+   * result from a CloudWatch event.
+   *
+   * @see https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/resource-based-policies-cwe.html#sqs-permissions
+   */
+  public bind(rule: events.IRule): events.RuleTargetProperties {
+    // deduplicated automatically
+    this.queue.grantSendMessages(new iam.ServicePrincipal('events.amazonaws.com',
+      {
+        conditions: {
+          ArnEquals: { "aws:SourceArn": rule.ruleArn }
+        }
+      })
+    );
+
+    const result = {
+      id: this.queue.node.id,
+      arn: this.queue.queueArn,
+      input: this.props.message,
+    };
+    if (this.queue.queueName.split('.').pop() === fifoQualifier && this.props.messageGroupId !== undefined) {
```

Comment:
Sorry, you can't do this.

If the queue object is created on the spot, `queueName` will not be a literal value. See the final section of this doc: https://docs.aws.amazon.com/cdk/latest/guide/lifecycle.html

You will just have to trust that if people pass a `messageGroupId` it will be for a FIFO queue.

Reply:
I didn't think about that: you are right. I encountered the same issue when I had to manually define a domain name [here](https://github.com/made2591/immutable.templates/blob/master/templates/static-website/lib/website-stack/storage-stack/storage-stack.ts) (line 111) and I ended up using the cdk.Fn.\* functions because I had an `undefined` before... In this case, I first tried to look at the object, but nothing helped me at the object level. Unfortunately, the only way I found was the *trick* of the name convention - that even it is formalized, tbh I didn't like. In any case yes, let's trust consumer.